src/Analyzer.JsonRuleEngine/Expressions/LeafExpression.cs

```diff
@@ -44,49 +38,29 @@ public LeafExpression(string resourceType, string path, LeafExpressionOperator @
         /// <summary>
         /// Evaluates this leaf expression's resource type and/or path, starting at the specified json scope, with the contained <c>LeafExpressionOperator</c>.
         /// </summary>
-        /// <param name="jsonScope">The json to evaluate.</param>
-        /// <returns>A <see cref="JsonRuleEvaluation"/> with zero or more results of the evaluation, depending on whether there are any/multiple resources of the given type,
-        /// and if the path contains any wildcards.</returns>
+        /// <param name="jsonScope">The json path to evaluate.</param>
+        /// <returns>A <see cref="JsonRuleEvaluation"/> with the result of the evaluation.</returns>
         public override JsonRuleEvaluation Evaluate(IJsonPathResolver jsonScope)
         {
-            if (jsonScope == null)
-            {
-                throw new ArgumentNullException(nameof(jsonScope));
-            }
-
-            List<IJsonPathResolver> scopesToEvaluate = new List<IJsonPathResolver>();
-
-            if (!string.IsNullOrEmpty(ResourceType))
-            {
-                scopesToEvaluate.AddRange(jsonScope.ResolveResourceType(ResourceType));
-            }
-            else
-            {
-                scopesToEvaluate.Add(jsonScope);
-            }
-
-            List<JsonRuleResult> jsonRuleResults = new List<JsonRuleResult>();
-            bool evaluationPassed = true;
-
-            foreach (var scope in scopesToEvaluate)
-            {
-                var leafScope = scope?.Resolve(Path);
-
-                foreach (var propertyToEvaluate in leafScope)
+            return EvaluateInternal(
+                jsonScope,
+                getResult: scope =>
                 {
-                    bool passed = Operator.EvaluateExpression(propertyToEvaluate.JToken);
-                    evaluationPassed &= passed;
-
-                    jsonRuleResults.Add(new JsonRuleResult
+                    var result = new JsonRuleResult()
                     {
-                        Passed = passed,
-                        JsonPath = propertyToEvaluate.Path,
+                        Passed = Operator.EvaluateExpression(scope.JToken),
+                        JsonPath = scope.Path,
                         Expression = this
-                    });
-                }
-            }
+                    };
+
+                    try
+                    {
+                        result.LineNumber = this.jsonLineNumberResolver.ResolveLineNumber(result.JsonPath);
+                    }
+                    catch (Exception) { }
```

Comment:
Can a test be added for this catch statement?

Reply:
I decided this was not necessary.  If this does ever throw an exception, it will be because of a bug in rule evaluation setup, not something wrong in `LeafExpression`.